packages/devtools_app/lib/src/inspector/inspector_screen.dart

```diff
@@ -372,45 +350,95 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
   }
 }
 
-class ErrorNavigator extends StatelessWidget {
-  const ErrorNavigator(
-      {Key key,
-      @required this.selectedErrorIndex,
-      @required this.errorCount,
-      @required this.onSelectedErrorIndexChanged})
-      : super(key: key);
+class ErrorNavigator extends StatefulWidget {
+  const ErrorNavigator({
+    Key key,
+    @required this.errors,
+    @required this.selectedNode,
+    @required this.selectedErrorIndex,
+  }) : super(key: key);
+
+  final LinkedHashMap<String, InspectableWidgetError> errors;
+  final ValueListenable<InspectorTreeNode> selectedNode;
+  final ValueNotifier<int> selectedErrorIndex;
 
-  final int selectedErrorIndex;
-  final int errorCount;
-  final void Function(int) onSelectedErrorIndexChanged;
+  @override
+  _ErrorNavigatorState createState() => _ErrorNavigatorState();
+}
+
+class _ErrorNavigatorState extends State<ErrorNavigator> with AutoDisposeMixin {
+  @override
+  void initState() {
+    super.initState();
+    widget.selectedNode?.addListener(_selectedNodeChanged);
+  }
+
+  @override
+  void didUpdateWidget(covariant ErrorNavigator oldWidget) {
+    if (oldWidget.selectedNode != widget.selectedNode) {
+      oldWidget.selectedNode?.removeListener(_selectedNodeChanged);
+      widget.selectedNode?.addListener(_selectedNodeChanged);
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    widget.selectedNode?.removeListener(_selectedNodeChanged);
+  }
+
+  void _selectedNodeChanged() {
+    final node = widget.selectedNode.value;
+    final inspectorRef = node?.diagnostic?.valueRef?.id;
+    // Check whether the node that was just selected has any errors associated
+    // with it.
+    var errorIndex = inspectorRef != null
+        ? widget.errors.keys.toList().indexOf(inspectorRef)
+        : null;
```

Comment:
why not -1 instead of null?

Reply:
The honest answer is that -1 made the line short enough that it unwrapped and seemed much less readable than this (and both did the same thing, since the -1 from `indexOf()` is also then converted to null below. Happy to change if you prefer though!